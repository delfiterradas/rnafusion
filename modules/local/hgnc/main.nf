process HGNC_DOWNLOAD {
    tag "hgnc"
    label 'process_low'

    conda "bioconda::gnu-wget=1.18"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/3b/3b54fa9135194c72a18d00db6b399c03248103f87e43ca75e4b50d61179994b3/data' :
        'community.wave.seqera.io/library/wget:1.21.4--8b0fcde81c17be5e' }"

    output:
    path "hgnc_complete_set.txt"        , emit: hgnc_ref
    path "HGNC-DB-timestamp.txt"        , emit: hgnc_date
    path "versions.yml"                 , emit: versions


    script:
    """
    wget --no-check-certificate https://storage.googleapis.com/public-download-files/hgnc/tsv/tsv/hgnc_complete_set.txt
    date +%Y-%m-%d/%H:%M  > HGNC-DB-timestamp.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        wget: \$(wget --version | head -1 | cut -d ' ' -f 3)
    END_VERSIONS
    """

    stub:
    """
    touch "hgnc_complete_set.txt"
    touch "HGNC-DB-timestamp.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        wget: \$(wget --version | head -1 | cut -d ' ' -f 3)
    END_VERSIONS
    """

}
