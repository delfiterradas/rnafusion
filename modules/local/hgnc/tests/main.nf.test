nextflow_process {

    name "Test Process HGNC_DOWNLOAD"
    script "../main.nf"
    process "HGNC_DOWNLOAD"

    test("Should download HGNC files") {

        when {
            process {
                """
                // This process doesn't have any inputs
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.hgnc_ref,
                    file(process.out.hgnc_date[0]).name,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("Should create stub files") {

        options "-stub"

        when {
            process {
                """
                // This process doesn't have any inputs
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.hgnc_ref,
                    file(process.out.hgnc_date[0]).name,
                    process.out.versions
                ).match() }
            )
        }
    }
}
