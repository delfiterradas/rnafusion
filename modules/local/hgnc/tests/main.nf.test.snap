{
    "Should create stub files": {
        "content": [
            [
                "hgnc_complete_set.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            "HGNC-DB-timestamp.txt",
            [
                "versions.yml:md5,a7b7ccbd9eda8036baf548cdf1cb6867"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T17:42:04.47884487"
    },
    "Should download HGNC files": {
        "content": [
            [
                "hgnc_complete_set.txt:md5,29571d88d1648e8764b70791df6a5d2d"
            ],
            "HGNC-DB-timestamp.txt",
            [
                "versions.yml:md5,a7b7ccbd9eda8036baf548cdf1cb6867"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T17:41:49.965512273"
    }
}